src/action.test.js

```diff
@@ -0,0 +1,94 @@
+import configureStore from 'redux-mock-store';
+
+import thunk from 'redux-thunk';
+
+import { fetchCategories, fetchRegions, fetchRestaurants } from './services/api';
+
+import {
+  setCategories,
+  loadCategories,
+  setRegions,
+  loadRegions,
+  loadRestaurants,
+  setRestaurants,
+} from './actions';
+
+const middleware = [thunk];
+
+const mockStore = configureStore(middleware);
+
+describe('async actions', () => {
+  const store = mockStore({});
+
+  describe('loadCategories', () => {
+    global.fetch = jest.fn(() => Promise.resolve({
+      json: () => Promise.resolve({ categories: [{ id: 1, name: '한식' }] }),
+    }));
+
+    beforeEach(() => {
+      fetch.mockClear();
+    });
+
+    it('should fetch categories data', async () => {
+      await store.dispatch(loadCategories());
+
+      const categories = await fetchCategories();
+
+      expect(store.getActions()[0]).toEqual(setCategories(categories));
+    });
+  });
+
+  describe('loadRegions', () => {
+    global.fetch = jest.fn(() => Promise.resolve({
+      json: () => Promise.resolve({ regions: [{ id: 1, name: '서울' }] }),
+    }));
+
+    beforeEach(() => {
+      fetch.mockClear();
+    });
+
+    it('should fetch regions data', async () => {
+      await store.dispatch(loadRegions());
+
+      const regions = await fetchRegions();
+
+      expect(store.getActions()[1]).toEqual(setRegions(regions));
+    });
+  });
+
+  describe('loadRestaurants', () => {
+    global.fetch = jest.fn(() => Promise.resolve({
+      json: () => Promise.resolve({ restaurants: [{ id: 1, name: '새마을 식당' }] }),
+    }));
+
+    beforeEach(() => {
+      fetch.mockClear();
+    });
+
+    it('should fetch restaurants data', async () => {
+      const categoryId = 1;
+
+      const regionName = '서울';
+
+      const stopLoad = 1;
+
+      await store.dispatch(loadRestaurants(categoryId, regionName, stopLoad));
```

Comment:
```suggestion
      await store.dispatch(loadRestaurants({ category, region });
```

위와 같이 한다면 훨씬 깔끔할 겁니다. 변수가 3개 이상이면 순서를 외우기도 어렵고, 오브젝트로 해결하는 게 좋습니다.
RORO 패턴을 벌써 잊으신 건 아니죵?

- https://www.freecodecamp.org/news/elegant-patterns-in-modern-javascript-roro-be01e7669cbd/

그나저나 stopLoad가 뭔지 모르겠네요...

Reply:
그렇네요ㅎㅎ 맞습니다~! 수정하겠습니다!
stopLoad는 category랑 region 이 선택된 상태에서 계속 fetching 을 해와서 network 탭을 보니 
![image](https://user-images.githubusercontent.com/60169820/97069001-c889fa80-1607-11eb-9fc8-f539e85585e9.png)
이렇게 되더라구요 그래서 stopfetch 가 맞겠네요. 아무튼 한번 fetch해오면 계속 못하게끔 막아줄려고 stopLoad 가 0 이면 fetch 못해오겠끔 설정해줬습니다.

Reply:
다른 방법도 찾아보겠습니다~!